File: src/handler.py

from typing import Dict

from helpers import mastodon, twitter, bluesky
from helpers.dynamodb import DynamoDBWrapper
from helpers.wikimedia import get_random_image

dynamodb = DynamoDBWrapper()

def post(file: Dict[str, any]) -> None:
  mastodon.post(file)
  twitter.post(file)
  bluesky.post(file)

def handler(_, __):
  file = get_random_image()
  dynamodb.record_post_to_table(file['pageid'], file['title'])
  post(file)

File: src/handler.ts
import {
  DynamoDBWrapper,
  getRandomImage,
  postToMastodon,
  postToTwitter,
} from './helpers';

const dynamodb = new DynamoDBWrapper();

const post = async (file: { pageid: number; title: string }) => {
  await Promise.all([postToMastodon(file), postToTwitter(file)]);
};

export const handler = async (_event: any, _context: any): Promise<void> => {
  const file = await getRandomImage();
  await dynamodb.recordPostToTable(file.pageid, file.title);
  await post(file);
};
